docs/extensiondev.rst

```diff
@@ -542,3 +542,229 @@ your HTTP requests::
+key areas that should be tested. For more exhaustive examples, you may want to
+take a look at the test cases that ship with Mopidy itself which covers
+everything from instantiating various controllers, reading configuration files,
+and simulating events that your extension can listen to.
+
+In general your tests should cover the extension definition, the relevant
+Mopidy controllers, and the Pykka backend and / or frontend actors that form
+part of the extension.
+
+Testing the extension definition
+--------------------------------
+Test cases for checking the definition of the extension should ensure that:
+
+- the extension provides a ``ext.conf`` configuration file containing the
+  relevant parameters with their default values,
+- that the config schema is fully defined, and
+- that the extension's actor(s) are added to the Mopidy registry on setup.
+
+An example of what these tests could look like is provided below::
+
+    def test_get_default_config(self):
+        ext = Extension()
+        config = ext.get_default_config()
+
+        self.assertIn('[my_extension]', config)
+        self.assertIn('enabled = true', config)
+        self.assertIn('param_1 = value_1', config)
+        self.assertIn('param_2 = value_2', config)
+        self.assertIn('param_n = value_n', config)
+
+    def test_get_config_schema(self):
+        ext = Extension()
+        schema = ext.get_config_schema()
+
+        self.assertIn('enabled', schema)
+        self.assertIn('param_1', schema)
+        self.assertIn('param_2', schema)
+        self.assertIn('param_n', schema)
+
+    def test_setup(self):
+        registry = mock.Mock()
+
+        ext = Extension()
+        ext.setup(registry)
+        calls = [mock.call('frontend', frontend_lib.MyFrontend),
+                 mock.call('backend',  backend_lib.MyBackend)]
+        registry.add.assert_has_calls(calls, any_order=True)
+
+
+Testing backend actors
+----------------------
+Backends can usually be constructed with a small mockup of the configuration
+file, and mocking the audio actor::
+
+    @pytest.fixture()
+    def config():
+        return {
+            'http': {
+                'hostname': '127.0.0.1',
+                'port': '6680'
+            },
+            'proxy': {
+                'hostname': 'host_mock',
+                'port': 'port_mock'
+            },
+            'my_extension': {
+                'enabled': True,
+                'param_1': 'value_1',
+                'param_2': 'value_2',
+                'param_n': 'value_n',
+            }
+        }
+
+    def get_backend(config):
+        return backend.MyBackend(config=config, audio=mock.Mock())
+
+
+You'll probably want to patch ``requests`` or any other web API's that you use
+to avoid any unintended HTTP requests from being made by your backend during
+testing::
+
+    from mock import patch
+    @mock.patch('requests.get',
+                mock.Mock(side_effect=Exception('Intercepted unintended HTTP call')))
+
```

Comment:
Should we mention [responses](https://pypi.python.org/pypi/responses) or [vcrpy](https://pypi.python.org/pypi/vcrpy) here?